using_restricted_array.rb

```diff
@@ -38,20 +71,38 @@ def sort(array, length)
 # constant, adds an element with 'SPECIAL_VALUE' in the end. Assumes the array
 # to be sorted in ascending order.
 def delete(array, length, value_to_delete)
-  puts "NOT IMPLEMENTED"
+  i = 0
+  length.times do
+    if value_to_delete == array[i]
+      array[i] = SPECIAL_VALUE

```

Comment:
Currently this may cause a SPECIAL_VALUE to be in the middle of valid values in the array (fragmentation). How can you update this method so all valid values are towards the end?